build-system/tasks/presubmit-checks.js

```diff
@@ -588,6 +588,9 @@ const forbiddenTerms = {
   '\\.defer\\(\\)': {
     message: 'Promise.defer() is deprecated and should not be used.',
   },
+  'assert\\(TAG': {
```

Comment:
There is a corner case obviously if someone decided to set the variable that they actually want to assert on as TAG. I think that is a fine corner case and they'll probably figure it out ok.

The other corner case is just if someone doesn't name their 'tag' TAG. This seems like a pretty standard nomenclature though, and this rule is better than nothing, so I like it as is.

Reply:
Yeah, I was concerned about the latter. We could modify assert() itself, so that it complains if it gets the wrong number of args...

Reply:
Actually, can't make assert() complain about wrong number of args (aside from if it gets zero args):
```
   * @param {T} shouldBeTrueish The value to assert. The assert fails if it does
   *     not evaluate to true.
   * @param {string=} opt_message The assertion message
   * @param {...*} var_args Arguments substituted into %s in the message.
```

Reply:
@erwinmombay @aghassemi Please review changes to this file. The remaining files are likely not relevant to you, though. Thanks!

Reply:
We have a more advanced regex for `#error`, `#info` and the like: https://github.com/jonkeller/amphtml/blob/4878eb1d42a18a91f7cd722e635fac328fac81a0/build-system/tasks/presubmit-checks.js#L487.

Let's update this one to:

```js
'(dev|user)\\(\\)\\.assert(Element|String|Number)?\\(\\s*([A-Z0-9-]+|[\'"`])'

Reply:
I like that, but now it's complaining about stuff like https://github.com/ampproject/amphtml/blob/master/src/custom-element.js#L230
I think we should change [A-Z0-9-]+ to [A-Z0-9-]+, (add comma at the end)

Reply:
Two more issues:
1. There's also a dev().assert(0, ...) in src/element-stub.js
I thought about changing 0 to false there, but seems like that should be allowed. So instead I switched [A-Z0-9-]+ to [A-Z][A-Z0-9-]*
2. There are a couple of dev().assert('prop' in obj, ...) so I removed the |[\'"`] altogether.

Reply:
Pushed, back to you @jridgewell